lib/facter/java_version.rb

```diff
@@ -11,6 +11,11 @@
 #
 # Notes:
 #   None
+osfamily = Facter.value(:osfamily)
+case osfamily
+when 'OpenBSD'
+  ENV['PATH'] = ENV['PATH'] + File::PATH_SEPARATOR + '/usr/local/jdk-1.7.0/jre/bin'
```

Comment:
@elyscape Agreed, Facter should not affect the ENV.

From looking at facter, there seems to be precedent to do what you want in an easier way:

```ruby
DEFAULT_SEARCH_PATHS = ['/sbin', '/usr/sbin']

  def search_paths
    ENV['PATH'].split(File::PATH_SEPARATOR) + DEFAULT_SEARCH_PATHS
  end

  def which(bin)
    if absolute_path?(bin)
      return bin if File.executable?(bin)
    else
      search_paths.each do |dir|
        dest = File.join(dir, bin)
        return dest if File.executable?(dest)
      end
    end
    nil
  end
```
Taken from https://github.com/puppetlabs/facter/blob/master/lib/facter/core/execution/posix.rb